pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -355,28 +357,35 @@ func (m *Multicluster) initializeCluster(cluster *multicluster.Cluster, kubeCont
 
 // checkShouldLead returns true if the caller should attempt leader election for a remote cluster.
 func (m *Multicluster) checkShouldLead(client kubelib.Client, systemNamespace string) bool {
+	var res bool
 	if features.ExternalIstiod {
-		namespace, err := client.Kube().CoreV1().Namespaces().Get(context.TODO(), systemNamespace, metav1.GetOptions{})
-		if err == nil {
+		b := backoff.NewExponentialBackOff(backoff.DefaultOption())
+		// context for both timeout and channel, whichever stops first, the context will be done
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		_ = b.RetryWithContext(ctx, func() error {
+			namespace, err := client.Kube().CoreV1().Namespaces().Get(context.TODO(), systemNamespace, metav1.GetOptions{})
```

Comment:
can the informer we already have be used? or we haven't started it at this point?

Reply:
So we will give up after 5s?

Reply:
yes, we haven't started at this point.

>So we will give up after 5s?

right

Reply:
Ok. Curious, have you run in to this issue and retry helped - When would that happen typically?

Reply:
If this happen, it is very serious. The mesh can fail to work

Reply:
Yes. I get that point.  That is why I was asking, in which case this can happen and did retries help in your case and if it is not helping - would it make sense to give up after 5s?

Reply:
Transient network error.  We do not retry forever until success is not to block bootstrap, it is like what we do with syncAll previously, even it is not done, after some time we will mark istiod ready.

Reply:
Here this is not perfect, what do you think?

Reply:
One of the problems I think with going ahead is we do not setup[ webhook patching ](https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/multicluster.go#L311) for remote clusters. And it fails silently till we get an error.  It is not caused by this PR but we should look at trying it in the background if we are not able to get namespace here. WDYT?

Reply:
IC, make sense to me. And also i will add some logs